src/ripple/app/main/GRPCServer.cpp

```diff
@@ -271,9 +310,37 @@ GRPCServerImpl::CallData<Request, Response>::getUsage()
             peer = clientIp;
         }
     }
-    boost::optional<beast::IP::Endpoint> endpoint =
-        beast::IP::Endpoint::from_string_checked(peer);
-    return app_.getResourceManager().newInboundEndpoint(endpoint.get());
+    auto res = beast::IP::Endpoint::from_string_checked(peer);
+    if (res)
+        return {res.get()};
+    else
```

Comment:
If the IP address can't be figured out and none is returned, then the port will become UNLIMITED, right, because no IP is set in the message.

Reply:
Right, we can add a user field to the protobuf messages to make sure the port does not become UNLIMITED when the software fails to figure out the value of a non-empty `client_ip`.